src/ui/simple/CSimpleEditBox.cpp

```diff
@@ -996,6 +996,14 @@ void CSimpleEditBox::OnLayerHide() {
     CSimpleEditBox::ClearKeyboardFocus(this, true);
 }
 
+void CSimpleEditBox::OnLayerShow() {
+    CSimpleFrame::OnLayerShow();
+
+    if (!this->IsCurrentFocus() && this->m_autoFocus) {
```

Comment:
This isn't quite right. It should be:
```suggestion
    if (!CSimpleEditBox::s_currentFocus && this->m_autoFocus) {
```

It's checking if there's no focus, not if the edit box isn't currently focused.